### PR TITLE
Make compression for io-hdfs configurable

### PR DESCRIPTION
**Pull request by LesTR**
Also added some tests + fixed typos

### CODE REVIEW

File: core/src/test/java/cz/o2/proxima/util/TestUtils.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import static org.junit.Assert.assertTrue;
+
+public class TestUtils {
+  /**
+   * Check if object is serializable
+   *
+   * @param object Object to check
+   * @throws IOException
+   */
+  public static void assertSerializable(Object object) throws IOException {
```
Comment (je-ik) [inline core/src/test/java/cz/o2/proxima/util/TestUtils.java]: This is cool. Can you reuse this please on other tests that test this? 
 - `StreamElementCoderTest.testCoderSerializable`
 - `ConfigRepositoryTest.testRepositoryNotSerializable` (actually tests the opposite, but that should not be a problem)
 - `DirectDataOperatorTest.testRepositoryNotSerializable` (dtto)
 - `cz.o2.proxima.direct.hbase.SerializationTest`
 - `cz.o2.proxima.direct.http.SerializableTest`

The modifications should be straightforward, if not, would you create issue to track it?

File: core/src/test/java/cz/o2/proxima/util/TestUtils.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+    try (ObjectOutputStream oos = new ObjectOutputStream(buffer)) {
+      oos.writeObject(object);
+    }
+    assertTrue(
+        String.format("Class '%s' isn't serializable.", object.getClass().getName()),
+        buffer.toByteArray().length > 0);
```
Comment (je-ik) [inline core/src/test/java/cz/o2/proxima/util/TestUtils.java]: Maybe we can recreate the object from the bytes and check they are `equal`?

File: core/src/test/java/cz/o2/proxima/util/TestUtils.java (hunk 1 above)
Comment (LesTR) [reply core/src/test/java/cz/o2/proxima/util/TestUtils.java]: fixed

File: core/src/test/java/cz/o2/proxima/util/TestUtils.java (hunk 2 above)
Comment (LesTR) [reply core/src/test/java/cz/o2/proxima/util/TestUtils.java]: fixed

File: direct/io-hbase-bindings/src/main/java/cz/o2/proxima/direct/hbase/HBaseLogObservable.java [hunk 3]
```diff
@@ -196,4 +196,20 @@ private Filter toFilter(List<AttributeDescriptor<?>> attributes) {
 
     return attrFilter;
   }
+
+  @Override
+  public int hashCode() {
+    return 69;
```
Comment (je-ik) [inline direct/io-hbase-bindings/src/main/java/cz/o2/proxima/direct/hbase/HBaseLogObservable.java]: Minor: this is not a prime, prime would have been better for this.

Comment (je-ik) [reply direct/io-hbase-bindings/src/main/java/cz/o2/proxima/direct/hbase/HBaseLogObservable.java]:
```suggestion
    return 73;
```

Comment (LesTR) [reply direct/io-hbase-bindings/src/main/java/cz/o2/proxima/direct/hbase/HBaseLogObservable.java]: fixed

**Pull request merged**